.github/scripts/validate-pr-number.js

```diff
@@ -0,0 +1,36 @@
+// @ts-check
+
+const { readFileSync } = require('node:fs');
+
+module.exports = main;
+
+/**
+ *
+ * @param {{filePath:string}} options
+ * @returns {number}
+ */
+function main(options) {
+  return validatePrNumber(options.filePath);
+}
+
+/**
+ *
+ * @param {string} filePath
+ * @returns {number}
+ */
+function validatePrNumber(filePath) {
+  try {
+    const content = readFileSync(filePath, 'utf-8').trim();
+    const prNumber = Number(content);
+
+    if (isNaN(prNumber) || !Number.isInteger(prNumber)) {
```

Comment:
nit: using `Number.isInteger` should enough since it also checks for `NaN`, making `isNaN` redundant.